src/opLog.ts

```diff
@@ -1,17 +1,59 @@
-export class OperationLog<State extends ReadonlyState, ReadonlyState, Operation> {
+class RingBuffer<T> {
+    private data: T[] = []
+    private next: number = 0
+
+    constructor(public size: number) { }
+
+    private wrap(idx: number, lim: number) {
+        while (lim && idx >= lim)
+            idx -= lim
+        return idx
+    }
+
+    add(t: T) {
+        this.data[this.next] = t
+
+        this.next = this.wrap(this.next + 1, this.size)
+    }
+
+    *[Symbol.iterator]() {
```

Comment:
Not supported in IE, but I guess that doesn't matter since this is arcade

Reply:
If we decide to support IE, this won't be hard to change